### PR TITLE
run e2e-openstack-extended test in self hosted runner

### PR DESCRIPTION
**Pull request by eslutsky**
this GH action will trigger  openstack extended tests in  a self hosted runner .

this PR consists of the following items:
-  [x] trigger the test  with a certain comment and report the job status on the PR.
-  [ ] provision self hosted fedora VM on ovirt and join as a runner to forklift gh repo.
    - [x] create a fedora template  with kind cluster +  gh-runner instance pre-configured.
            each template is configured to work with specific `org/repo`
    - [x] create a VM from the template (limited to 1  VM per template)
    - [ ] allow running several VM instances from same template - generate runner-id and reconfigure the runner 
-  [x] run the extended on the self hosted runner.
-  [x] remove the runner VM
-  [x] make sure only [codeowner](https://github.com/kubev2v/forklift/blob/main/.github/CODEOWNERS#L5) in `tests/ can trigger the job using  comment.
-  [ ] scale-out - support multiple  concurrent runs (up to maximum)

### CODE REVIEW

**Comment by eslutsky** [comment]
example [PR](https://github.com/eslutsky/forklift/pull/4) that show this functionality

File: .github/workflows/create-new-pr.yml
Comment (ahadas) [inline .github/workflows/create-new-pr.yml:24]: isn't it too strict? I'm ok with letting developers that are not code-owners to initiate the test, how about checking it in a similar way to [how it's done for oVirt's tests](https://github.com/oVirt/ovirt-system-tests/blob/master/.github/workflows/ost.yaml#L37-L38)?

File: .github/workflows/run-osp-extended-tests.yml
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:5]:
```suggestion
      description: 'run e2e openstack extended tests on a self-hosted runner'
```

File: .github/workflows/run-osp-extended-tests.yml
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:43]: the other comment on this check should have been placed here but also applies there to the message by the bot

File: .github/workflows/run-osp-extended-tests.yml
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:65]: remove then?

File: .github/workflows/run-osp-extended-tests.yml
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:84]: why do we need docker hub?

File: .github/workflows/run-osp-extended-tests.yml
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:126]:
```suggestion
            const name = '${{ github.workflow }}';
```

File: .github/workflows/run-osp-extended-tests.yml
Comment (eslutsky) [reply .github/workflows/run-osp-extended-tests.yml:84]: it needed for pulling kind images , with out this line it fails on docker Pull rate limits.

File: .github/workflows/create-new-pr.yml
Comment (eslutsky) [reply .github/workflows/create-new-pr.yml]: refactored

File: .github/workflows/run-osp-extended-tests.yml
Comment (ahadas) [reply .github/workflows/run-osp-extended-tests.yml:80]: ack

File: .github/workflows/create-new-pr.yml
Comment (ahadas) [inline .github/workflows/create-new-pr.yml:17]: do we really need to checkout the forklift repo here?

File: .github/workflows/run-osp-extended-tests.yml
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:43]: will be changed before merging?

File: .github/workflows/run-osp-extended-tests.yml
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:50]: same here

File: .github/workflows/run-osp-extended-tests.yml
Comment (ahadas) [inline .github/workflows/run-osp-extended-tests.yml:105]: why doing this rather than just `if: github.event_name == 'issue_comment'`? (here and below)

**Review by ahadas — COMMENTED** [review]
looks good overall, few minor comments inside

File: .github/workflows/create-new-pr.yml
Comment (eslutsky) [reply .github/workflows/create-new-pr.yml]: apparently not, fixed.

File: .github/workflows/run-osp-extended-tests.yml
Comment (eslutsky) [reply .github/workflows/run-osp-extended-tests.yml]: done

File: .github/workflows/run-osp-extended-tests.yml
Comment (eslutsky) [reply .github/workflows/run-osp-extended-tests.yml]: done

File: .github/workflows/run-osp-extended-tests.yml
Comment (eslutsky) [reply .github/workflows/run-osp-extended-tests.yml:103]: we want this action to be triggered regardless of the previous step result (succees or failure),
according to the [docs](https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions) 
```
You can use the following status check functions as expressions in if conditionals. A default status check of success() is applied unless you include one of these functions.
```
Where “one of these functions” refers to:
- sucess()
- always()
- cancelled()
- failure()

Comment (ahadas) [reply .github/workflows/run-osp-extended-tests.yml:103]: got it

**Review by ahadas — APPROVED** [review]

**Pull request merged**
